### ISSUE TITLE
Unfortunately error message again when calling up channels

### ISSUE THREAD

**Issue by bjoern-vh** [lekma/plugin.video.invidious#80]
Since the last weekend I cannot cal channels any more. I have to xxxxx out the domain because it is my self hosted Invidious instance but the same happens with the public instances. 

I get this error:

`
2023-04-03 16:07:10.915 T:894      info <general>: [plugin.video.invidious] <service.session> request: https://tube.xxxxxxx.de/api/v1/channels/UC4zcMHyrT_xyWlgy5WGpFFQ/videos?page=1&region=DE
2023-04-03 16:07:11.962 T:894      info <general>: [plugin.video.invidious] <service.session> request: https://tube.xxxxxxx.de/api/v1/channels/UC4zcMHyrT_xyWlgy5WGpFFQ?region=DE
2023-04-03 16:07:12.266 T:1027    error <general>: GetDirectory - Error getting plugin://plugin.video.invidious/?action=channel&authorId=UC4zcMHyrT_xyWlgy5WGpFFQ
2023-04-03 16:07:12.279 T:862     error <general>: CGUIMediaWindow::GetDirectory(plugin://plugin.video.invidious/?action=channel&authorId=UC4zcMHyrT_xyWlgy5WGpFFQ) failed
2023-04-03 16:07:12.288 T:1028    error <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the followingerror<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'TypeError'>
                                                   Error Contents: list indices must be integers or slices, not str
                                                   Traceback (most recent call last):
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/plugin.py", line 226, in <module>
                                                       dispatch(*argv)
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/plugin.py", line 222, in dispatch
                                                       InvidiousPlugin(url, int(handle)).dispatch(**parseQuery(query))
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/plugin.py", line 27, in dispatch
                                                       super(InvidiousPlugin, self).dispatch(**kwargs)
                                                     File "/storage/.kodi/addons/script.module.iapc/lib/iapc/tools/plugin.py", line 66, in dis
patch
                                                       return action(**kwargs)
                                                              ^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/script.module.iapc/lib/iapc/tools/plugin.py", line 33, in wra
pper
                                                       raise error
                                                     File "/storage/.kodi/addons/script.module.iapc/lib/iapc/tools/plugin.py", line 30, in wra
pper
                                                       success = func(self, **kwargs)
                                                                 ^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/plugin.py", line 103, in channel
                                                       return self.addDirectory(client.channel(**kwargs), "video", **kwargs)
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^
                                                     File "/storage/.kodi/addons/plugin.video.invidious/lib/invidious/client.py", line 88, in
channel
                                                       data["videos"], continuation=data.get("continuation"),
                                                       ~~~~^^^^^^^^^^
                                                   TypeError: list indices must be integers or slices, not str
                                                   -->End of Python script error report<--
`

I hope it's just a small thing and you can solve the problem.

Many thanks in advance :-)

**Comment by lekma** [comment]
> Since the last weekend I cannot cal channels any more. I have to xxxxx out the domain because it is my self hosted Invidious instance but the same happens with the public instances.
> 

some public instances respond correctly (right now):

- [https://invidious.projectsegfau.lt/api/v1/channels/UC4zcMHyrT_xyWlgy5WGpFFQ/videos?page=1&region=DE](https://invidious.projectsegfau.lt/api/v1/channels/UC4zcMHyrT_xyWlgy5WGpFFQ/videos?page=1&region=DE)
- [https://invidious.tiekoetter.com/api/v1/channels/UC4zcMHyrT_xyWlgy5WGpFFQ/videos?page=1&region=DE](https://invidious.tiekoetter.com/api/v1/channels/UC4zcMHyrT_xyWlgy5WGpFFQ/videos?page=1&region=DE)

can you check in a browser what your instance returns?

(I'm abroad right now and don't have access to my computer, so I'll do what I can, but won't be able to commit anything till next week, unfortunately)

**Comment by lekma** [comment]
also what version of invidious server are you running?

**Comment by bjoern-vh** [comment]
You have put me on the right track. My server version was probably too old. After an update it works now again.

I have checked the other instances again. These give a different error message and do not run. On another computer in the browser everything is displayed correctly. 

I had installed and activated OpenVPN in my Kodi for testing purposes. However, I did not get it uninstalled cleanly again. I also have problems with other addons. My own Invidious instance is running now though. If I have time, I will reinstall Kodi.

For now, the issue has been resolved. Thanks for your immediate feedback and sorry for the false alarm :-)
